### ISSUE TITLE
Implement Partial Register Dataflow

### ISSUE THREAD

**Issue by joshwatson** [Vector35/binaryninja-api#873]
It looks like there have been some major dataflow regressions with partial register accesses. Given the following x86:
```
mov al, 1
mov ah, 2
```
The following LLIL is emitted:
```
>>> current_llil[0]
<il: al = 1>
>>> current_llil[1]
<il: ah = 2>
```
However, the dataflow incorrectly returns 'undetermined` for query the partial register values that would be affected by this:
```
>>> current_llil[1].get_reg_value_after('ax')
<undetermined>
>>> current_llil[1].get_reg_value_after('ah')
<undetermined>
>>> current_llil[1].get_reg_value_after('al')
<undetermined>
```
However, if `ah` is moused over, the tooltip is correct:
![screen shot 2017-11-16 at 9 52 39 am](https://user-images.githubusercontent.com/5834130/32907373-a14db8d6-cab4-11e7-9a47-093af6e48506.png)
*But*, if the full width register is moused over, the tooltip is incorrect:
![screen shot 2017-11-16 at 9 52 01 am](https://user-images.githubusercontent.com/5834130/32907395-b4a2cb60-cab4-11e7-8ee7-35cef55c9de5.png)
The full width register `eax` _should_ in this case return 'undetermined', because `ah` and `al` do not sign extend or zero extend the full width register.

**Comment by bpotchik** [comment]
Renamed title since this is not a regression. The GetRegisterValue system never had an implementation for partial register value access, it's currently a TODO in the code. Partial register access is implemented for the PossibleValue dataflow system. Looking into this now.

**Comment by bpotchik** [comment]
![image](https://user-images.githubusercontent.com/29309548/35653597-1837d588-06b7-11e8-8f98-6c72cace2afc.png)

**Comment by bpotchik** [comment]
Fixed in build 1.1.1051-dev.
Note, if a register is partially defined the possible value API must be used to query the constant value. Partial constants are propagated at this level.
